src/sprites/entity.py

```diff
@@ -69,17 +73,33 @@ def __init__(
     def get_state(self):
         self.state = 'walk' if self.direction else 'idle'
 
-    def get_facing_direction(self):
+    def get_facing_direction(self, direction: tuple[float, float] = None):
+        """
+        :param direction: Direction to use.
+                          If it is not set, uses self.direction instead
+        """
         # prioritizes vertical animations, flip if statements to get horizontal
         # ones
-        if self.direction.x:
-            self.facing_direction = 'right' if self.direction.x > 0 else 'left'
-        if self.direction.y:
-            self.facing_direction = 'down' if self.direction.y > 0 else 'up'
+        if not direction:
+            direction = self.direction
```

Comment:
Isn’t self.direction a Vector2, normally?

Reply:
Yes, my intention behind those lines was to make Entity.get_facing_direction work with an optional direction parameter, that accepts a tuple to avoid unnecessary overhead. However, I think that it might be a better solution to extract that functionality into a separate function in support.py, and make Entity.get_facing_direction call that helper function with self.direction instead, right?

Reply:
Good idea